plugins/modules/mount.py

```diff
@@ -251,46 +251,35 @@ def is_fspath_mounted(module, mount_dir, mount_over_dir):
 
     if mount_over_dir:
         # when mounting NFS, make sure we check for the NFS
-        # client mount point
-        cmd = "lsfs -l %s" % mount_over_dir
+        fs_name = mount_over_dir
     elif mount_dir:
-        cmd = "lsfs -l %s" % mount_dir
+        fs_name = mount_dir
     else:
         # should not happen
         result['msg'] = "Unexpected module FAILURE: one of the following is missing: "
         result['msg'] += ','.join(['mount_dir', 'mount_over_dir'])
         module.fail_json(**result)
 
+    cmd = "df"
     rc, stdout, stderr = module.run_command(cmd)
-
     if rc != 0:
-        return None
-
-    if rc == 0:
-        # Get the fs_name of the Filesystem
-        # check if it is listed in the first column of df command
-        ln = stdout.splitlines()[1]
-        fs_name = ln.split()[0]
-        cmd = "df"
-        rc, stdout, stderr = module.run_command(cmd)
-        if rc != 0:
-            result['msg'] = "Failed to get the filesystem name. Command '%s' failed." % cmd
-            result['cmd'] = cmd
-            result['rc'] = rc
-            result['stdout'] = stdout
-            result['stderr'] = stderr
-            module.fail_json(**result)
-
-        fdirs = []
-        if stdout:
-            for ln in stdout.splitlines()[1:]:
-                fdirs.append(ln.split()[0])
-        for fdir in fdirs:
-            found = re.search(fs_name, fdir)
-            if found:
-                return True
-
-        return False
+        result['msg'] = "Failed to get the filesystem name. Command '%s' failed." % cmd
+        result['cmd'] = cmd
+        result['rc'] = rc
+        result['stdout'] = stdout
+        result['stderr'] = stderr
+        module.fail_json(**result)
+
+    fdirs = []
+    if stdout:
+        for ln in stdout.splitlines()[1:]:
+            fdirs.append(ln.split()[-1])
+    for fdir in fdirs:
+        found = re.search(fs_name, fdir)
+        if found:
+            return True
+
+    return False
```

Comment:
when return False, it means fs_name could not be found from "df" command, should we log cmd, rc, stdout, stderr here?

Reply:
we don't have to in here, this is a helper function that just checks
if the filesystem is already mounted or not

it will be the calling function's responsibility to decide what to do with the information
they get from this helper function (logging cmd, rc, stdout, stderr if needed)

Reply:
the only time we fail in this helper function is if the df command itself fails.
we the log like you say

however if the df command run successfully, we can then start to figure out
if the specified FS is mounted or not
True - mounted
False - not mounted